src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/parsing/MailboxParsingService.java

```diff
@@ -18,21 +19,29 @@ public class MailboxParsingService {
 
     private final MailboxParser mailboxParser;
 
-    public MailboxParsingService(MailboxFileProvider mailboxRepository,
-            MailboxParser mailboxParser) {
-        this.mailboxFileProvider = mailboxRepository;
-        this.mailboxParser = mailboxParser;
+    private final AuthorizationService authorizationService;
+
+
+  public MailboxParsingService(MailboxFileProvider mailboxRepository, MailboxParser mailboxParser,
+      AuthorizationService authorizationService) {
+    this.mailboxFileProvider = mailboxRepository;
+    this.mailboxParser = mailboxParser;
+    this.authorizationService = authorizationService;
+  }
+
+  public Mailbox getMailboxOfAddress(User user) throws MailboxParsingServiceException {
+
+    if (!authorizationService.authorize(user.email(), user.password())) {
+      throw new MailboxParsingServiceException("User not authorized");
     }
 
```

Comment:
It is not a Exception of the ParsingService. It is an Exception of the Authorization Service. You could just rewrite it like this:

```suggestion
  public Mailbox getMailboxOfAddress(User user) throws MailboxParsingServiceException, AuthorizationException{
      authorizationService.authorize(user);
  }
```

And in the authorizationService it should throw an AuthorizationException("User not authorized");